client/src/http.rs

```diff
@@ -10,11 +10,21 @@ pub use http::{Method, Response, StatusCode};
 ///
 /// The order of builder methods invocation is not strict. There is no guarantee that builder user calls
 /// all methods. Only [`RequestBuilder::new`] is the required one.
-pub trait RequestBuilder {
-    /// Entrypoint - create a new builder with specified method and URL.
+pub trait RequestBuilder
+where
+    Self: Sized,
+{
     #[must_use]
+    #[deprecated]
+    /// Entrypoint - create a new builder with specified method and URL.
     fn new(method: Method, url: impl AsRef<str>) -> Self;
 
+    /// Entrypoint - create a new builder with specified method and URL.
+    ///
+    /// # Errors
+    /// If either the `method` or `url` are invalid.
```

Comment:
I am not sure whether `method` could be invalid in any case because of `Method` type which follows the "parse, not validate" pattern. It cannot be constructed without validation.

Reply:
Maybe what's meant here is that e.g. the server may or may not support methods such as `CONNECT`, and in the latter case calling the function with it would be invalid (or rather disallowed).